File: tests/fixtures/typescript/slop_catch.ts
function risky(): void {}
function doWork(): void {}

function emptyCatch() {
  try {
    risky();
  } catch (e) {} // expect: SLOP005
}

function logOnlyCatch() {
  try {
    doWork();
  } catch (err) { // expect: SLOP005
    console.log(err);
  }
}

function commentOnlyCatch() {
  try {
    doWork();
  } catch (e) { // expect: SLOP005
    // left blank
  }
}

function multiConsoleCatch() {
  try {
    doWork();
  } catch (e) { // expect: SLOP005
    console.warn(e);
    console.error(e);
  }
}

File: tests/fixtures/rust/slop_catch.rs
use std::fmt;

#[derive(Debug)]
struct MyError;

impl fmt::Display for MyError {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "boom")
    }
}

fn risky() -> Result<i32, MyError> {
    Err(MyError)
}

fn empty_arm() {
    match risky() {
        Err(e) => {}, // expect: SLOP005
        Ok(v) => println!("{}", v),
    }
}

fn comment_only_arm() {
    match risky() {
        Err(e) => { // expect: SLOP005
            // left blank
        },
        Ok(v) => println!("{}", v),
    }
}
